### PR TITLE
Add script run every Sunday to backup kmq db

### PR DESCRIPTION
**Pull request by taahamahdi**

### CODE REVIEW

File: src/scripts/backup-kmq-database.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { execSync } from "child_process";
+import { friendlyFormattedDate } from "../helpers/utils";
+
+const databaseBackupDir = `${process.env.AOIMIRAI_DUMP_DIR}/kmq_backup`;
+
+function backupKmqDatabase() {
+    if (!fs.existsSync(databaseBackupDir)) {
+        fs.mkdirSync(databaseBackupDir);
+    }
+    execSync(`mysqldump -u ${process.env.DB_USER} -p${process.env.DB_PASS} -h ${process.env.DB_HOST} --routines kmq > ${databaseBackupDir}/kmq_backup_${friendlyFormattedDate(new Date())}.sql`);
```
Comment (Brainicism) [inline src/scripts/backup-kmq-database.ts:11]: This should be async to not block the main thread

File: src/helpers/management_utils.ts [hunk 2]
```diff
@@ -169,18 +170,26 @@ export function registerIntervals() {
...
     });
 
     // every monday at 7am UTC => 2am EST
-    schedule.scheduleJob("0 7 * * 1", async () => {
+    schedule.scheduleJob("0 7 * * MON", async () => {
         logger.info("Performing regularly scheduled AoiMirai database seed");
         await seedAndDownloadNewSongs();
     });
 
+    // every sunday at 12am UTC => 7pm EST
```
Comment (Brainicism) [inline src/helpers/management_utils.ts:186]: Prod runs on UTC, this is 7am UTC

File: src/helpers/management_utils.ts [hunk 3]
```diff
@@ -169,18 +170,26 @@ export function registerIntervals() {
...
+    // everyday at 12am UTC => 7pm EST
     schedule.scheduleJob("0 0 * * *", async () => {
         const serverCount = state.client.guilds.size;
         storeDailyStats(serverCount);
         reloadFactCache();
     });
 
     // every monday at 7am UTC => 2am EST
-    schedule.scheduleJob("0 7 * * 1", async () => {
+    schedule.scheduleJob("0 7 * * MON", async () => {
```
Comment (Brainicism) [inline src/helpers/management_utils.ts:181]: Keep it as '1' to be consistent

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/scripts/backup-kmq-database.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+}
+
```
Comment (Brainicism) [inline src/scripts/backup-kmq-database.ts:13]: Add main method here that runs the backup incase we want to do it manually

**Review by Brainicism — CHANGES_REQUESTED** [review]

File: src/helpers/management_utils.ts (hunk 2 above)
Comment (taahamahdi) [reply src/helpers/management_utils.ts:186]: Whoops, I meant to replace the 7 with 0

File: src/helpers/management_utils.ts [hunk 5]
```diff
@@ -181,6 +183,13 @@ export function registerIntervals() {
         await seedAndDownloadNewSongs();
     });
 
+    // every sunday at 12am UTC => 7pm saturday EST
+    schedule.scheduleJob("0 0 * * 0", async () => {
```
Comment (Brainicism) [inline src/helpers/management_utils.ts:187]: Offset this by a little bit to make sure it's after the `storeDailyStats`

File: src/scripts/backup-kmq-database.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import fs from "fs";
+import { exec } from "child_process";
+import { friendlyFormattedDate } from "../helpers/utils";
+
+const databaseBackupDir = `${process.env.AOIMIRAI_DUMP_DIR}/kmq_backup`;
+
+async function backupKmqDatabase() {
```
Comment (Brainicism) [inline src/scripts/backup-kmq-database.ts:7]: You can make this return a Promise that resolves on `exec`'s callback, so we can still await the call

File: package.json [hunk 7]
```diff
@@ -15,7 +15,8 @@
     "lint": "npx eslint . --ext .js,.ts",
     "docker-db": "cd docker/ && docker-compose up -d db",
     "docker-dev": "cd docker/ && START_TYPE=dev docker-compose up --build kmq",
-    "docker-prod": "cd docker/ && START_TYPE=prod docker-compose up --build kmq"
+    "docker-prod": "cd docker/ && START_TYPE=prod docker-compose up --build kmq",
+    "backup": "ts-node src/scripts/backup-kmq-database"
```
Comment (Brainicism) [inline package.json:19]: Add script for import here

**Review by Brainicism — CHANGES_REQUESTED** [review]

**Review by Brainicism — APPROVED** [review]

**Pull request merged**